LocalPayment/src/main/java/com/braintreepayments/api/localpayment/LocalPaymentClient.kt

```diff
@@ -0,0 +1,258 @@
+package com.braintreepayments.api.localpayment
+
+import android.content.Context
+import android.net.Uri
+import androidx.annotation.VisibleForTesting
+import com.braintreepayments.api.BrowserSwitchFinalResult
+import com.braintreepayments.api.BrowserSwitchOptions
+import com.braintreepayments.api.core.AnalyticsEventParams
+import com.braintreepayments.api.core.BraintreeClient
+import com.braintreepayments.api.core.BraintreeException
+import com.braintreepayments.api.core.BraintreeRequestCodes
+import com.braintreepayments.api.core.Configuration
+import com.braintreepayments.api.core.ConfigurationException
+import com.braintreepayments.api.datacollector.DataCollector
+import com.braintreepayments.api.sharedutils.Json
+import org.json.JSONException
+import org.json.JSONObject
+import java.util.Locale
+
+/**
+ * Used to integrate with local payments.
+ */
+class LocalPaymentClient @VisibleForTesting internal constructor(
+    private val braintreeClient: BraintreeClient,
+    private val dataCollector: DataCollector,
+    private val localPaymentApi: LocalPaymentApi
+) {
+    /**
+     * Used for linking events from the client to server side request
+     * In the Local Payment flow this will be a Payment Token/Order ID
+     */
+    private var payPalContextId: String? = null
+
+    /**
+     * Initializes a new [LocalPaymentClient] instance
+     *
+     * @param context an Android Context
+     * @param authorization a Tokenization Key or Client Token used to authenticate
+     * @param returnUrlScheme a custom return url to use for browser and app switching
+     */
+    constructor(
+        context: Context,
+        authorization: String,
+        returnUrlScheme: String?
+    ) : this(BraintreeClient(context, authorization, returnUrlScheme))
+
+    @VisibleForTesting
+    internal constructor(braintreeClient: BraintreeClient) : this(
```

Comment:
Can we collapse this constructor into the primary constructor by setting default values for `dataCollector` and `localPaymentApi`?